File: 2482-difference-between-ones-and-zeros-in-row-and-column/2482-difference-between-ones-and-zeros-in-row-and-column.cpp
class Solution {
public:
    vector<vector<int>> onesMinusZeros(vector<vector<int>>& arr) {
        vector<pair<int,int>>row(arr.size(),{0,0}),col(arr[0].size(),{0,0});
        for(int i=0;i<arr.size();i++){
            for(int j=0;j<arr[0].size();j++){
                if(arr[i][j]==1)row[i].first++,col[j].first++;
                else row[i].second++,col[j].second++;
            }            
        }
        vector<vector<int>>ans;
        for(int i=0;i<arr.size();i++){
            for(int j=0;j<arr[0].size();j++){
                arr[i][j]=row[i].first +col[j].first -row[i].second -col[j].second;
            }
        }
        return arr;
    }
};

File: 2482-difference-between-ones-and-zeros-in-row-and-column/2482-difference-between-ones-and-zeros-in-row-and-column.py
class Solution:
    def onesMinusZeros(self, arr: List[List[int]]) -> List[List[int]]:
        R, C = len(arr), len(arr[0])
        row, col = [0] * R, [0] * C
        for r in range(R):
            for c in range(C):
                offset = 2 * arr[r][c] - 1 
                row[r] += offset
                col[c] += offset
        return [[row[r] + col[c] for c in range(C)] for r in range(R)]
